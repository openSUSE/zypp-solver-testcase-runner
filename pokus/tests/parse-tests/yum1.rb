#!/usr/bin/ruby


# Generate a YUM repository and remove the file 'primary.xml'.


require 'zypptools/lib/arch'
require 'zypptools/lib/package'
require 'zypptools/output/output'
include ZYppTools

require '../../src/testsuite'


packages = Array.new()

pkg = Package.new("test")
pkg.arch = "ia64"
pkg.rpmsize = 42
pkg.version = "1.2"
pkg.release = 1
pkg.summary = "A pure test package"
packages.push(pkg)

path = Testsuite.write_repo(:yum, packages)

FileUtils.rm(path + "/repodata/primary.xml")


Testsuite.set_arch("ia64")

begin
    pool = Testsuite.read_repo("file://" + path)
rescue ZYppException => e
    puts "ZYppException caught"
    puts e.to_s
else
    puts "Oh, no exception caught"
    exit 1
end

