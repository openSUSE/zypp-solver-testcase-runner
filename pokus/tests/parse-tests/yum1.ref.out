ZYppException caught
History:
 - Can't provide ./repodata/primary.xml : Media Exception
 - File ./repodata/primary.xml not found on media: file:/tmp/tmp.abcdefghij
Valid metadata not found at specified URL(s)
